src/units/attack_type.hpp

```diff
@@ -54,7 +55,27 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	int num_attacks() const { return num_attacks_; }
 	double attack_weight() const { return attack_weight_; }
 	double defense_weight() const { return defense_weight_; }
-	const config &specials() const { return specials_; }
+	const ability_vector &specials() const { return specials_; }
+
+	/**
+	 * Returns a vector of all specials of the given type, unlike get_specials it
+	 * doesn't check whether  the specials are active.
```

Comment:
I think all of these functions should be renamed to make it more clear which ones check whether the specials are active. Also, the function above this one needs a Doxygen comment.

* `specials()` &rarr; `get_all_specials()` – returns all specials, not checking whether they're active
* `specials(std::string)` &rarr; `get_all_specials_of_type(std::string)` – returns specials with a particular tag, not checking whether they're active
* `get_specials(std::string)` &rarr; `get_active_specials_of_type(std::string)` – returns active specials with a particular tag
* `weapon_specials()` &rarr; `get_active_specials()` – not exactly sure if this is one of them, but if it in fact returns all active specials as the name seems to imply, then it should be renamed as mentioned, and if it returns all specials without checking for whether they're active, then it's redundant with `specials()` and should be removed. If it's some third thing, then further discussion is probably needed.

Reply:
hmm probably, i'm hoping that i can get rid a at least one of them, and will probably do this last then.